### ISSUE TITLE
avg_mean and/or avg_var in BatchNormalization not working?

### ISSUE THREAD

**Issue by benvcutilli** [chainer/chainer#8604]
Hi,

I successfully trained my neural network with chainer.links.BatchNormalization (I have gotten good loss, it trains faster as expected). However, when switching to chainer.global_config.train = False, the accuracy falls to the point at which it is almost like I didn't train the network at all. My network is a Sigmoid-based convolutional one, with (as the Batch Norm paper
recommends) the batch normalization layers taking the input of the dot product and outputting to the activation function. I've tried many decay values (0.9, the default of 0.99, 0.999, 0.2, and 0.0, the last of which was intended to act as though the test image would fit within the distribution of the batch that was last trained, which IMO is a reasonable assumption.

Running my test code (that processes batches of images) with the train mode on shows the performance I desire. The code that has issues is virtually identical to the test code, with the only difference between the two being that I pass in a single image of dimension (1, depth, height, width), which should work to be seen as a batch. Also, networks without batch normalization have worked fine. To eliminate this being an issue with parameters on the GPU being serialized incorrectly to be used on the CPU, I ran the test code with chainer.global_config.train on on both GPU and CPU, and it worked fine. Turning it off results in the same bad performance (although I'm not sure if this was on GPU or CPU).

I don't want to post too much of my code, so here are a few snippets (which all code in my, well, code uses, so it should eliminate the chance that this issue is due to multiple reimplementations of the same code being different). Hopefully this fulfills the code snippet requirement:

Example initialization:
`self.a_normalize = chainer.links.BatchNormalization(decay=<whatever>, axis=(0, 2, 3))`
Usage:
`volume     = self.a_normalize(self.a(volume))     # self.a is the convolutional layer`
Trained network loading:
```
def recreateNetwork(networkPath, settingsPath):
   model         = None
   modelFilename = pathlib.Path(networkPath).parts[-1]
    
   handle        = open(settingsPath)
   settings      = json.load(handle)
   handle.close()

   model = <I initialize a new model here>
   chainer.serializers.load_npz(networkPath, model)
    
   return model
```

And the network is used by treating it as a callable.

I've looked for help/reported bugs on Google, looked through the bug reports here, and have seen nothing. I know this could be me using it wrong, but I feel as though I've pretty much exhaustively looked into this issue, so there's a good chance that this is indeed a bug. Here is the output of python -c "import chainer; chainer.print_runtime_info()", as requested:

  Platform: Linux-3.10.0-1160.24.1.el7.x86_64-x86_64-with-glibc2.2.5
  Chainer: 7.7.0
  ChainerX: Not Available
  NumPy: 1.20.2
  CuPy:
    CuPy Version          : 7.8.0
    CUDA Root             : /usr/local/cuda
    CUDA Build Version    : 10000
    CUDA Driver Version   : 11030
    CUDA Runtime Version  : 10000
    cuBLAS Version        : 10000
    cuFFT Version         : 10000
    cuRAND Version        : 10000
    cuSOLVER Version      : (10, 0, 0)
    cuSPARSE Version      : 10000
    NVRTC Version         : (10, 0)
    cuDNN Build Version   : None
    cuDNN Version         : None
    NCCL Build Version    : None
    NCCL Runtime Version  : None
    CUB Version           : None
    cuTENSOR Version      : None
  iDeep: Not Available

Thanks! If any of this doesn't make sense, please let me know.

**Comment by benvcutilli** [comment]
Also, I should point out that when I try to try a `decay` value of `0.0`, the network is kinda close in accuracy, but not really (it doesn't come that close to what would be expected given that the loss is so good during training; this probably isn't a case of overfitting as I'm testing, for now, on the same images that I've trained on).

**Comment by benvcutilli** [comment]
Using the recommended 1.17 (well, 1.17.3) version of NumPy doesn't seem to help.

**Comment by benvcutilli** [comment]
I wrote my own variant of Batch Normalization, and it appears to be having similar issues both evaluating on the network that is left in memory right after training as well as one loaded from disk (aka deserialized). I think this means that there isn't an issue with Chainer's BatchNormalization, but rather an issue with my network (or dataset).
